docs/decisions/0003-jwt-usage.rst

```diff
@@ -0,0 +1,125 @@
+0003: JWT usage
+###############
+
+Status
+******
+
+**Draft**
+
+Context
+*******
+
+Json Web Tokens (JWT) are a way for authenticating users in web
+applications. The server generates a signed token that encodes the
+user's identity and any relevant claims, which is then sent to the
+client. The client includes this token in subsequent requests, allowing
+the server to verify the user's identity without needing to maintain
+session state.
+
+Because the tokens are cryptographically signed, they can be efficiently
+verified by the server without requiring database queries, which brings
+performance and scalability benefits.
+
+In addition to storing user identity, JWT tokens can also include
+additional claims, such as user roles or permissions, which can be used
+for authorization (AuthZ) purposes.
+
+JWT is the `recommended way to authenticate with the Open edX REST API
+<https://docs.openedx.org/projects/edx-platform/en/latest/how-tos/use_the_api.html>`_,
+however, it's not widely used for AuthZ across the platform. One
+exception is the `edx-rbac <https://github.com/openedx/edx-rbac>`_
+library, which was a previous attempt to implement role-based access
+control in the platform.
+
+The edx-rbac library adds role information to the `"roles" claim in the
+JWT token
+<https://github.com/openedx/edx-rbac/blob/master/docs/how_to_guide.rst>`_
+however, this is not widely used across the platform, being mostly used
+on enterprise modules.
+
+Given the new AuthZ project efforts, which seek to unify and simplify
+AuthZ on Open edX, there is an opportunity to re-evaluate the use of JWT
+tokens for Authorization purposes.
+
+Possible Methods
+================
```

Comment:
This is not actually important but the bullets in this section are nested too far and in the rendered view they view as quotes (https://openedx-authz--34.org.readthedocs.build/en/34/decisions/0003-jwt-usage.html#possible-methods). Probably not actually worth fixing, but worth paying attention to for more user-facing documentation.